### ISSUE TITLE
Minimum number of characters to trigger autocomplete

### ISSUE THREAD

**Issue by daitau** [jobisoft/TbSync#213]
First, thank you for the awesome addon.

I have 2 questions:

1. Is there a reason to restrict autocomplete to at least 4 characters in abAutoComplete.js? Is it to reduce the number of server-client transactions? I was a little bit confused when I test the addon and think I did not set up properly (I only entered 2 characters and nothing happened.) I wonder if it should be made a configurable option?
 
2. I would like to use [MRC Compose addon](https://addons.thunderbird.net/en-US/thunderbird/addon/mrc-compose/) with TbSync but seems it cannot make use of the autocomplete provided by TbSync. I am not sure if it is related to (1) or not.

**Comment by jobisoft** [comment]
1. Most EAS servers do not send a response if the search word is less than 3 chars. I thus hardcoded that limit to reduce traffic.

2. I do not know that add-on. What is the actual error?
a) just by installing that add-on, TbSync auto-completion is not working anymore
b) that add-on is adding its own auto-complete fields to Thunderbird and there TbSync is not adding its search results
c) something else

**Comment by daitau** [comment]
1. I see. Thank you for the explanation.

2. That add-on actually changed the recipient fields to resemble modern email UI like gmail or Outlook (i.e. one field contains comma-delimited email address(es) instead of one row for each recipient.) The problem I encountered is that the entries from TbSync are not included in the autocomplete list there, so (b) seems to be the problem. I have a feeling that TbSync results are not included because they are not actual entries in the addressbook (but it is not considered as an LDAP server either so there are no  mechanism to wait for result.) I haven't read the code to confirm this though. For now I need the autocomplete more so I stopped using that add-on at the moment.

**Comment by jobisoft** [comment]
TbSync adds its own auto complete handle to the original fields. It sounds like the other AddOn is overwriting the handles with defaults.

**Comment by jobisoft** [comment]
Let me rephrase that: The other add on could grab all registered auto complete handles from the original fields and add those to its own new fields, instead of just using the defaults.

**Comment by jobisoft** [comment]
Can I close this?

**Comment by daitau** [comment]
Yes, thank you again for the explanation.
